src/core/annotation.js

```diff
@@ -1097,7 +1097,14 @@ class WidgetAnnotation extends Annotation {
   }
 
   async save(evaluator, task, annotationStorage) {
-    if (this.data.fieldValue === annotationStorage[this.data.id]) {
+    const value = annotationStorage[this.data.id];
+    if (
+      this.data.fieldValue === value ||
```

Comment:
Nit: For improved readability, please change this to `value === this.data.fieldValue ||` such that all of the comparisons are sorted consistently in one `if` condition.